testsuite/test-runner/src/main/java/org/wildfly/core/testrunner/WildFlyRunner.java

```diff
@@ -184,34 +184,50 @@ public void testRunFinished(Result result) throws Exception {
             Security.insertProviderAt(ELYTRON_PROVIDER, 0);
             providerInstalled = true;
         }
+        RuntimeException setupException = null;
         try {
             if (automaticServerControl) {
-                runSetupTasks();
+                setupException = runSetupTasks();
             }
         } catch (AssumptionViolatedException e) {
             assumptionFailedInServerSetup = e;
         }
-        super.run(notifier);
+        if (setupException == null) {
+            super.run(notifier);
+        }
```

Comment:
Before this change, if an AssumptionViolatedException gets thrown, the `super.run(notifier);` was executed. Now, we are introducing a change here. I'm not familiar with JUInit runners, are there any side effects?

Reply:
IMO, it won't introduce a change here. With the change, the `super.run(notifier)` will also be executed when AssumptionViolatedException was thrown from `runSetupTasks()`, the `setupException` is `null` in that case.